internal/install/discovery/os_windows_validator.go

```diff
@@ -0,0 +1,62 @@
+package discovery
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+)
+
+type OsWindowsValidator struct{}
+
+var (
+	windowsVersionNoLongerSupported = "This version of Windows is no longer supported"
+	windowsNoVersionMessage         = "Failed to identified a valid version of Windows"
+)
+
+func NewOsWindowsValidator() *OsWindowsValidator {
+	validator := OsWindowsValidator{}
+
+	return &validator
+}
+
+func (v *OsWindowsValidator) Execute(m *types.DiscoveryManifest) string {
+	if m.OS != "windows" {
+		return ""
+	}
+
+	versions := strings.Split(m.PlatformVersion, ".")
```

Comment:
Not a blocker by any means, but the [`go-semvery`](https://github.com/coreos/go-semver) library has a nice API for these kinds of checks for things like `ensureMinimumVersion`.

Reply:
Good to know, I did struggle a bit with the golang regexp library yesterday, to the point I realize I didn't need a library to split on a "." :) 
But yeah, good to know about this library I guess.